src/system_identity.rs

```diff
@@ -30,15 +39,183 @@ type Base64PublicKey = String; // For L2 System Identity.
+        };
+        let expected_client_id = cli_input.system_identity_input.client_id.to_owned();
+        let expected_name = cli_input.name.to_owned();
+
+        let mut token_retriever_http_client = MockHttpClient::new();
+        // 1. Sign the token
+        token_retriever_http_client
+            .expect_send()
+            .once()
+            .withf({
+                let env = cli_input.environment.clone();
+                move |req| {
+                    req.method().eq(&http::Method::POST)
+                        && req.uri().eq(&env.token_renewal_endpoint())
+                        && req
+                            .headers()
+                            .get("Content-Type")
+                            .is_some_and(|v| v == "application/json")
+                }
+            })
+            .returning(|_| {
+                let response = http::Response::builder()
+                    .status(200)
+                    .body(
+                        serde_json::to_string(&serde_json::json!({
+                            "access_token": "some-access-token",
+                            "expires_in": 3600,
+                            "token_type": "Bearer",
+                        }))
+                        .unwrap()
+                        .as_bytes()
+                        .to_vec(),
+                    )
+                    .unwrap();
+                Ok(response)
+            });
+
+        let mut iam_client_http_client = MockHttpClient::new();
+        iam_client_http_client
+            .expect_send()
+            .once()
+            .withf({
+                let env = cli_input.environment.clone();
+                move |req| {
+                    req.method().eq(&http::Method::POST)
+                        && req.uri().eq(&env.identity_creation_endpoint())
+                        && req
+                            .headers()
+                            .get("Content-Type")
+                            .is_some_and(|v| v == "application/json")
+                }
+            })
+            .returning(move |_| {
+                let response = http::Response::builder()
+                    .status(200)
+                    .body(
+                        serde_json::to_string(&serde_json::json!({
+                          "data": {
+                            "systemIdentityCreate": {
+                              "clientId": format!("{}-client-id", ctx.name),
+                              "name": ctx.name,
+                              "publicKey": String::from_utf8_lossy(&[1u8, 2u8, 3u8]), // "cHVibGljS2V5QmFzZTY0RW5jb2RlZFN0cmluZw==",
+                              "id": "some-granted-id",
+                              "organizationId": "org-id",
+                              // "clientSecret": "c2VjcmV0LWNsaWVudC1zZWNyZXQtc2VjcmV0",
+                              // "credentialExpiration": "2023-10-01T00:00:00Z"
```

Comment:
I was the one who suggested to parametrize this test. I see now more clearly that we can't really parametrize it, right? Because the return type here should be different.

If the IAMClient didn't use the TokenRetriever, we could easily test the IAMClient by passing a "random-access-token" and avoid mocking. Another point to think about when we refactor at the end, if we decide to refactor.